pkg/operator/ceph/object/spec.go

```diff
@@ -1130,3 +1151,80 @@ func GetHostnameFromEndpoint(endpoint string) (string, error) {
 
 	return hostname, nil
 }
+
+func (c *clusterConfig) getRGWFailureDomain() string {
+	poolSpecs := make([]cephv1.PoolSpec, 0)
+	poolSpecs = append(poolSpecs, c.store.Spec.DataPool)
+	poolSpecs = append(poolSpecs, c.store.Spec.MetadataPool)
+	return topology.GetMinimumFailureDomain(poolSpecs)
+}
+
+func nodeTopologyInjectorContainer(imageName string) v1.Container {
+	return v1.Container{
+		Name:    "node-topology-injector",
+		Image:   imageName,
+		Command: []string{"rook"},
+		Args:    []string{"inject-node-topology"},
+		Env: []v1.EnvVar{
+			k8sutil.NodeEnvVar(),
+		},
+	}
+}
+
+func getNodeLabelEnvContainerVolumeAndMount() (v1.Volume, v1.VolumeMount) {
+	volName := "node-env"
+	v := v1.Volume{Name: volName, VolumeSource: v1.VolumeSource{EmptyDir: &v1.EmptyDirVolumeSource{}}}
+	m := v1.VolumeMount{Name: volName, MountPath: "/tmp"}
+	return v, m
+}
+
+func (c *clusterConfig) getRGWContainerCMD(rgwConfig *rgwConfig, args *[]string) []string {
+	// TODO: minimum ceph version should be tentacles
+	if rgwConfig.ReadReplicaPolicy == localizedReadReplicaPolicy && c.clusterInfo.CephVersion.IsAtLeastTentacle() {
+		*args = append(*args, cephconfig.NewFlag("crush location", fmt.Sprintf("\"%s\"", c.getRGWCrushLocation(rgwConfig))))
+		// TODO: this call should be made at the end of `makeDaemonContainer` method when all the arguments have been initialized.
+		normalizeArgsForScript(args)
+		rgwCmd := "source /tmp/env* && radosgw " + strings.Join(*args, " ")
```

Comment:
Can we give the temp file a specific name? It's not good practice to source files with globs

```suggestion
		rgwCmd := "source /tmp/env-name && radosgw " + strings.Join(*args, " ")
```

Reply:
used `node-env` as a specific name for the file.